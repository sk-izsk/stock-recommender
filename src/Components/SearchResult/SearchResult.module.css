.searchResult {
  border: 1px solid cadetblue;
  border-radius: 5px;
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  -ms-border-radius: 5px;
  -o-border-radius: 5px;
  padding: 16px;
  position: absolute;
  z-index: 10;
  background-color: white;
  width: auto;
}

.stockPrice {
  display: flex;
}

.name {
  margin: 8px;
}

.date {
  margin: 8px;
}

.amount {
  margin: 8px;
}

.recommendationContainer {
  margin: 8px;
}

.buy {
  background-color: yellow;
  padding: 8px;
  border-radius: 5px;
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  -ms-border-radius: 5px;
  -o-border-radius: 5px;
}

.hold {
  background-color: red;
  padding: 8px;
  border-radius: 5px;
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  -ms-border-radius: 5px;
  -o-border-radius: 5px;
}

.sell {
  background-color: green;
  padding: 8px;
  border-radius: 5px;
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  -ms-border-radius: 5px;
  -o-border-radius: 5px;
}

.button {
  background-color: cadetblue;
  border: 0;
  padding: 0px 16px;
  cursor: pointer;
  border-radius: 5px;
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  -ms-border-radius: 5px;
  -o-border-radius: 5px;
}

.socialMedia {
  display: flex;
  flex-direction: column;
}

.socialData {
  margin: 8px;
}
