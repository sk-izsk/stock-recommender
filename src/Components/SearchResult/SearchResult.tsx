import React from "react";
import { Data, getRecommendation, StockPrice } from "../../utils";
import { SocialMediaCount } from "../../utils/socialMediaCountGenerator";
import styles from "./SearchResult.module.css";

interface Props {
  stockPrice?: StockPrice[];
  socialMedia?: SocialMediaCount[];
  handleAddedStockList: (name: string, date: string, amount: number) => void;
}

export const SearchResult: React.FC<Props> = ({
  stockPrice,
  socialMedia,
  handleAddedStockList,
}) => {
  return (
    <div className={styles.searchResult}>
      {stockPrice?.map(
        (stock: StockPrice, index: number) =>
          stock.date &&
          stock.amount && (
            <div className={styles.stockPrice} key={index}>
              <div className={styles.name}>Name: {stock.name}</div>
              <div className={styles.date}>date: {stock.date}</div>
              <div className={styles.amount}>price: $ {stock.amount}</div>
              <div className={styles.recommendationContainer}>
                Recommendation:{" "}
                <span className={styles[getRecommendation(stock.amount)]}>
                  {getRecommendation(stock.amount)}
                </span>
              </div>
              <button
                onClick={(e) => {
                  e.preventDefault();
                  handleAddedStockList(
                    stock.name as string,
                    stock.date as string,
                    stock.amount as number
                  );
                }}
                className={styles.button}
              >
                Buy
              </button>
            </div>
          )
      )}
      {socialMedia?.map(
        (data: SocialMediaCount, index: number) =>
          data.data && (
            <div className={styles.socialMedia} key={index}>
              <h3> Platform: {data.name}</h3>
              {data.data.map((count: Data, id: number) => (
                <div className={styles.socialData} key={id}>
                  {count.date} : Post {count.amount}
                </div>
              ))}
            </div>
          )
      )}
    </div>
  );
};
