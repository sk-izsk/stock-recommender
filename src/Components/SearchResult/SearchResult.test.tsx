import { render } from "@testing-library/react";
import React from "react";
import { SearchResult } from "./SearchResult";

const mockProps = {
  handleAddedStockList: jest.fn(),
  stockPrice: [
    {
      name: "APPL",
      date: "29-03-21",
      amount: 950,
    },
  ],
  socialMedia: [
    {
      name: "APPL",
      data: [
        {
          date: "29-03-21",
          amount: 70,
        },
      ],
    },
  ],
};

test("App renders correctly", () => {
  const { asFragment } = render(<SearchResult {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
