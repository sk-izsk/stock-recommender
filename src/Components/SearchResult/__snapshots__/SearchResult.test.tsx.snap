// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="searchResult"
  >
    <div
      class="stockPrice"
    >
      <div
        class="name"
      >
        Name: APPL
      </div>
      <div
        class="date"
      >
        date: 29-03-21
      </div>
      <div
        class="amount"
      >
        price: $ 950
      </div>
      <div
        class="recommendationContainer"
      >
        Recommendation: 
        <span
          class="sell"
        >
          sell
        </span>
      </div>
      <button
        class="button"
      >
        Buy
      </button>
    </div>
    <div
      class="socialMedia"
    >
      <h3>
         Platform: APPL
      </h3>
      <div
        class="socialData"
      >
        29-03-21 : Post 70
      </div>
    </div>
  </div>
</DocumentFragment>
`;
